covfee/client/tasks/continuous_annotation/index.tsx

```diff
@@ -41,66 +41,212 @@ const ContinuousAnnotationTask: React.FC<Props> = (props) => {
+  // annotationsDataMirror is undefined, or empty.
+  useEffect(() => {
+    if (annotationsDataMirror === undefined) {
+      dispatch(actions.setSelectedAnnotationIndex(-1))
+    } else if (annotationsDataMirror.length === 0) {
+      dispatch(actions.setSelectedAnnotationIndex(-1))
+    } else if (selectedAnnotationIndex === -1) {
+      dispatch(actions.setSelectedAnnotationIndex(0))
+    }
+  }, [selectedAnnotationIndex, annotationsDataMirror])
+
+  //-------------------- Video playback -------------------- //
+
   // this is a custom dispatch function provided by Covfee
-  const dispatch = useDispatch()
 
-  // we read the state using useSelector
-  const mediaPaused = useSelector<State, boolean>((state) => state.mediaPaused)
-  const active_annotation = useSelector<State, string>(
-    (state) => state.active_annotation
-  )
-  const active_participant = useSelector<State, string>(
-    (state) => state.active_participant
-  )
-  const [showingGallery, setShowingGallery] = useState(false)
+  const numberOfVideoFrames = 5 // TODO: Initialize me on video load
+
+  /******************************************************/
+  /***** Logic to create a dummy video playback *********/
+  /******************************************************/
+  // This is temporary, just to test the annotation flow,
+  // and pushing back and forth annotations to the server
+  // through the API endpoints
+  const DUMMY_VIDEO_NUMBER_OF_FRAMES: number = numberOfVideoFrames
+  const [frameNumber, setFrameNumber] = useState(0)
+  useEffect(() => {
+    let videoPlaybackInterval: NodeJS.Timeout | undefined
+
+    if (isAnnotating) {
+      setFrameNumber(0)
+      videoPlaybackInterval = setInterval(() => {
+        // Increment the seconds state every second
+        setFrameNumber((prevFrameNumber) => prevFrameNumber + 1)
+      }, 33)
+    }
+
+    // Clean up the interval when the component unmounts or when the timer stops
+    return () => clearInterval(videoPlaybackInterval)
+  }, [isAnnotating]) // Run the effect whenever isRunning changes
+
+  useEffect(() => {
+    if (frameNumber >= DUMMY_VIDEO_NUMBER_OF_FRAMES) {
+      console.log("Stopping the video playback")
+      setIsAnnotating(false)
+      handleVideoEnded()
+    }
+  }, [frameNumber])
+  /******************************************************/
+  /******************************************************/
 
   const my_video = {
     type: "video",
     url: "https://mdn.github.io/learning-area/html/multimedia-and-embedding/video-and-audio-content/rabbit320.mp4",
   }
 
-  // We prepare the annotations options
-  const annotations_menu_items: MenuProps["items"] = [
+  const selectFirstAvailableAnnotationIndexBasedOnParticipantName = (
+    participant: string
+  ) => {
+    if (annotationsDataMirror === undefined) {
+      return
+    }
+    // Finds the first annotation in the spec that has the participant
+    const first_annotation_index_for_participant =
```

Comment:
Will this work with several people annotating in parallel? I don't quite understand the code but it looks like the index might end up being duplicated.

Reply:
I'm not sure I understand the use case of several people annotating in parallel, please let me know. But to explain further  what this is doing: the explicit list of participants separate from the list of annotations was removed. This is due to the new introduction of the participant column in the table related to annotation data points (data point = participant + annotation category, eg. "Participant_1",  "Speaking")

Thus now we have a list "annotationsDataMirror" comprising of "tuples" (id, Participant, Category, data_json, ...) at each element of the list. Given this breaks the guarantee that we have the cartesian product of participant, category permutations, instead the anchor, for the user interface, became to select a participant and then, for a given participant, extract from the list those entries containing said participant, and thus create the annotation "Category" options for the user interface.

So, what happens when the user changes in the dropdown the participant, which annotation category should be selected, well... this logic says: the first one available.